integration_tests/grpc_rest/dclupgrade/helpers.go

```diff
@@ -0,0 +1,337 @@
+
+	// Check upgrade is approved
+	approvedUpgrade, err := GetApprovedUpgrade(suite, proposeUpgradeMsg.Plan.Name)
+	require.NoError(suite.T, err)
+	require.Equal(suite.T, proposeUpgradeMsg.Creator, approvedUpgrade.Creator)
+	require.Equal(suite.T, proposeUpgradeMsg.Plan, approvedUpgrade.Plan)
+
+	// Get all proposed upgrades
+	proposedUpgrades, err = GetProposedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.NotContains(suite.T, proposedUpgrades, *proposedUpgrade)
+
+	// Get all approved upgrades
+	approvedUpgrades, err := GetApprovedUpgrades(suite)
+	require.NoError(suite.T, err)
+	require.Contains(suite.T, approvedUpgrades, *approvedUpgrade)
+}
+
+/* Error cases */
+
+func ProposeUpgradeByNonTrustee(suite *utils.TestSuite) {
+	// Alice and Bob are predefined Trustees
+	aliceName := testconstants.AliceAccount
+	aliceKeyInfo, err := suite.Kr.Key(aliceName)
+	require.NoError(suite.T, err)
+	aliceAccount, err := test_dclauth.GetAccount(suite, aliceKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	bobName := testconstants.BobAccount
+	bobKeyInfo, err := suite.Kr.Key(bobName)
+	require.NoError(suite.T, err)
+	bobAccount, err := test_dclauth.GetAccount(suite, bobKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	for _, role := range []dclauthtypes.AccountRole{
+		dclauthtypes.CertificationCenter,
+		dclauthtypes.Vendor,
+		dclauthtypes.NodeAdmin,
+	} {
+		// register new account without Trustee role
+		nonTrusteeAccountName := utils.RandString()
+		nonTrusteeAccount := test_dclauth.CreateAccount(
+			suite,
+			nonTrusteeAccountName,
+			dclauthtypes.AccountRoles{role},
+			int32(tmrand.Uint16()+1),
+			aliceName,
+			aliceAccount,
+			bobName,
+			bobAccount,
+			testconstants.Info,
+		)
+
+		// try to add proposeUpgradeMsg
+		proposeUpgradeMsg := NewMsgProposeUpgrade(nonTrusteeAccount.Address, utils.RandString(), 100000, utils.RandString())
+		_, err = suite.BuildAndBroadcastTx([]sdk.Msg{proposeUpgradeMsg}, nonTrusteeAccountName, nonTrusteeAccount)
+		require.Error(suite.T, err)
+		require.ErrorIs(suite.T, err, sdkerrors.ErrUnauthorized)
+	}
+}
+
+func ApproveUpgradeByNonTrustee(suite *utils.TestSuite) {
+	// Alice and Bob are predefined Trustees
+	aliceName := testconstants.AliceAccount
+	aliceKeyInfo, err := suite.Kr.Key(aliceName)
+	require.NoError(suite.T, err)
+	aliceAccount, err := test_dclauth.GetAccount(suite, aliceKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	bobName := testconstants.BobAccount
+	bobKeyInfo, err := suite.Kr.Key(bobName)
+	require.NoError(suite.T, err)
+	bobAccount, err := test_dclauth.GetAccount(suite, bobKeyInfo.GetAddress())
+	require.NoError(suite.T, err)
+
+	// propose upgrade
+	proposeUpgradeMsg := NewMsgProposeUpgrade(aliceAccount.Address, utils.RandString(), 100000, utils.RandString())
+	_, err = suite.BuildAndBroadcastTx([]sdk.Msg{proposeUpgradeMsg}, aliceName, aliceAccount)
+	require.NoError(suite.T, err)
+
+	for _, role := range []dclauthtypes.AccountRole{
```

Comment:
Integration tests are time-consuming, so why not to grant all these roles just to one user?